### ISSUE TITLE
2fa (from App) Login fails "incorrect code"

### ISSUE THREAD

**Issue by mcguffin** [jotta/jotta-cli-issues#87]
**jotta-cli release (`jotta-cli version`):**
```
-------------------------------------------
jottad executable : /usr/bin/jottad
jottad appdata    : /var/lib/jottad
jottad version    : 0.5.15025
jotta-cli version : 0.5.15025
-------------------------------------------
```
**Description of problem:**
The two factor athentication code from app does not work.
```
$ jotta-cli login
[...]
accept license (yes/no): y
username: <MY_EMAIL_ADDRESS>
password: 
two factor authentication code: XXXXXX
Incorrect code
OK
```

**jotta-cli status (`jotta-cli status`):**
```
ERROR  Not logged in
```

**Relevant logs for the issue (`cat ~/.jottad/jottabackup.log`)**
```
No such file or directory
```

**Additional info:**
 - TOTP-App: [FreeOTP+](https://f-droid.org/en/packages/org.liberty.android.freeotpplus/)
 - OS: both Raspbian GNU/Linux 9.9 (stretch), macOS
 - 2fa from SMS works in `jotta-cli login`
 - runnig `jotta-cli` as root had no effect
 - entering the code in both formats `XXX-XXX` and `XXXXXX` did not help either
 - Auth-Codes from App work in both macOS Desktop client and on the jottacloud.com website

**Comment by mcguffin** [comment]
Supplement: Trying to set up a connection with app provided OTP in [rclone](https://rclone.org/) resulted in the following error message:
```
$ rclone config
[...]
Your Jottacloud password is only required during setup and will not be stored.
password: 
2019/08/03 13:53:24 Failed to get resource token: HTTP error 400 (400 Bad Request) returned body: "{\"error\":\"invalid_client_or_otp_missing_or_wrong\",\"error_description\":\"Client authentication failed (e.g., unknown client, no client authentication included, unsupported authentication method, or the resource owner has enabled OTP and the otp code is not supplied in the password grant request or the supplied otp code is wrong).\"}"
```
(Please note: rcrypt didn't ask for an OTP. The error occured right after I entered my password)

Using an SMS-OTP everything worked as expected.

**Comment by Kimbsen** [comment]
We are aware of the problem with OTP and jotta-cli. This happened because we moved to a new auth system internally which supports OTP, but jotta-cli is still on the old auth-system and only supports sms 2fa.

We will release a new jotta-cli that supports the new system shortly. Probably a week or two. Just need to finish some internal testing.

**Comment by mcguffin** [comment]
Thanks for letting me know! Looking forward to the next release.

**Comment by Kimbsen** [comment]
We just released 0.6 which changes the login flow. You now need to generate a personal login token when logged in on jottacloud.com and use this in place of username/password when logging into jotta-cli.

**Comment by mcguffin** [comment]
Awesome, Thanks!
